cmd/juju/application/bundle.go

```diff
@@ -351,6 +367,27 @@ func (h *bundleHandler) resolveCharmsAndEndpoints() error {
 	return nil
 }
 
+func (h *bundleHandler) resolveRelativeCharmPath(app, path string) (string, error) {
+	if filepath.IsAbs(path) {
+		return path, nil
+	}
+
+	absPath, err := filepath.Abs(filepath.Join(h.bundleDir, path))
+	if err != nil {
+		return "", errors.NotValidf("charm path %q in application %q", path, app)
+	}
+
+	if _, err = os.Stat(absPath); err == nil {
+		return absPath, nil
+	}
+
+	if os.IsNotExist(err) {
+		return "", errors.Errorf("charm path in application %q does not exit: %s", app, path)
```

Comment:
"does not exist"

Reply:
Nice catch. I will force-push to fix this.